library/argument/src/ap_option_with_parameter.e

```diff
@@ -93,9 +95,13 @@ feature -- Access
 	parameter: detachable G
 			-- Last value give to the option
 		require
-			was_found: was_found
+			was_found: was_found or has_default_value
 		do
-			Result := parameters.last
+			if parameters.is_empty then
```

Comment:
I think it is inconsistent that parameters is empty, while parameter returns a value (the default one). Alternatively, one could put the default value also into parameters.